worlds/soe/options.py

```diff
@@ -280,3 +280,19 @@ class SoEOptions(PerGameCommonOptions):
     trap_chance_hud:       TrapChanceHUD
     trap_chance_ohko:      TrapChanceOHKO
     progression_balancing: SoEProgressionBalancing
+
+    @property
+    def trap_chances(self) -> Iterator[TrapChance]:
+        for field in fields(self):
+            option = getattr(self, field.name)
+            if isinstance(option, TrapChance):
+                yield option
+
+    @property
+    def flags(self) -> str:
+        flags = ''
+        for field in fields(self):
+            option = getattr(self, field.name)
+            if isinstance(option, (EvermizerFlag, EvermizerFlags)):
+                flags += getattr(self, field.name).to_flag()
```

Comment:
I didn't notice before merge,
but wouldn't this be?
`flags += option.to_flag()`
so you don't have to `getattr` again?

Reply:
You are kind of right, I saw that after `git add`, but it needs a type ignore, or two separate instance() calls, or yet another type abstraction, so I decided to not bother at the moment.

Reply:
Yeah, I didn't notice that those were mixins.

This is how I would do it:
```python3
        for field in fields(self):
            option = getattr(self, field.name)
            if isinstance(option, (EvermizerFlag, EvermizerFlags)):
                assert isinstance(option, Option)
                flags += option.to_flag()
        return flags
```

Reply:
Interesting. That seems like a super easy fix, but I don't understand it (and neither does pycharm 2023.2.3)

Reply:
I'm guessing pycharm doesn't understand intersection types?

mypy and pyright see the intersection types:
`option: <subclass of EvermizerFlag and Option> | <subclass of EvermizerFlags and Option>`

Reply:
Oh, I see. the assert creates "typings" that fulfil the individual protocol. Took me a while to see why this works.
And yeah, pycharm seems to think those are Option now, rather than a combined pseudo type.

Reply:
Fixed in #2724